### PR TITLE
fix: preserve negative zero in std.round

### PR DESCRIPTION
**Pull request by He-Pin**
## Motivation

`std.round(-0)` returned positive zero in sjsonnet, while go-jsonnet and jrsonnet preserve the negative zero sign. This matters for programs that observe signed zero through functions such as `std.atan2`.

## Modification

- Preserve the original input when `std.round` receives either `+0` or `-0`.
- Keep the existing positive and negative rounding branches for non-zero numbers.
- Add a regression test that checks the sign of rounded zero values.

## Result

| Case | go-jsonnet v0.22.0 | jrsonnet 0.5.0-pre99 | sjsonnet before | sjsonnet after |
| --- | --- | --- | --- | --- |
| `std.round(-0)` | `-0` | `-0` | `0` | `-0` |
| `std.round(0)` | `0` | `0` | `0` | `0` |

`std.atan2(std.round(-0), -1)` now observes the negative-zero sign.

## Risks

- This intentionally changes materialized output for the exact `std.round(-0)` case.
- Other rounding behavior is intended to remain unchanged.

### CODE REVIEW

File: sjsonnet/src/sjsonnet/Val.scala [hunk 1]
```diff
@@ -78,12 +78,16 @@ final class LazyExpr(
...
-      r
+      try {
+        val r = ev.visitExpr(exprOrVal.asInstanceOf[Expr])(scope)
+        exprOrVal = r // cache result
+        scope = null.asInstanceOf[sjsonnet.ValScope] // allow GC
+        ev = null // sentinel: marks as computed
+        r
+      } finally {
+        // Allow re-evaluation on retry; same error will be thrown from exprOrVal
+        evaluating = false
```
Comment (He-Pin) [inline sjsonnet/src/sjsonnet/Val.scala:88]: ouch , it should be extracted

**Comment by stephenamar-db** [comment]
please fix PR.

**Comment by He-Pin** [comment]
@stephenamar-db fixed

**Review by stephenamar-db — APPROVED** [review]
